test/opds_parser_test.ts

```diff
@@ -55,8 +50,8 @@ describe("OPDSParser", () => {
         if (error) {
           done(error);
         } else {
-          let promise: Promise<OPDSFeed> = parser.parse(data);
-          promise.then((result) => {
+          let promise: Promise<OPDSEntry | OPDSFeed> = parser.parse(data);
```

Comment:
This tripped me up the most. Seems both possible types have to be declared but the actual type of the object has to be one of them (next line).

Reply:
`parser.parse` can return either. I'm kind of surprised the next line works.

Reply:
Since it’s based on compile time, it doesn’t know it could be until the following line where it’s explicitly told what it is. I’m surprised as well but it makes sense.